### DISCUSSION TITLE
How to get the token returned by a login?

### DISCUSSION THREAD

**Discussion by mcampbell** [prescottprue/cypress-firebase#376] [Q&A] [answered]
I'm considering using cypress to test our API layer, which requires the auth token from firebase auth to be on a header.  After using `cy.login()` I'm not sure how to go about getting that token.  Doing it from `httpie` + `bash` is something like:

`TOKEN=$(http post 'https://identitytoolkit.googleapis.com/v1/accounts:signInWithPassword' key=="$PROJECT_KEY" email="$EMAIL" password="$PASSWORD" returnSecureToken=true | jq '.idToken')`

Which I can use in my subsequent `httpie` calls.

Is there an equivalent in cypress-firebase?

**Comment by prescottprue** [comment, marked as answer]
`cy.login` uses a custom task which is added by cypress-firebase called `createCustomToken` which returns the token after calling firebase-admin internally - you can call like so:

```js
cy.task('createCustomToken', 'SomeUid')
 .then((token) => {
   // call whatever is needed here
  })
```

Let me know if I'm misunderstanding
